### DISCUSSION TITLE
Block "alias"

### DISCUSSION THREAD

**Discussion by juniusfree** [athensresearch/athens#1001] [Feature Requests]
**EDIT: aliases were implemented this way: https://github.com/athensresearch/athens/discussions/1326#discussioncomment-923738.**

Problem: update/paraphrase a block content without losing the original context.

Related Issue:
1. https://github.com/athensresearch/athens/issues/582

Inspired by:
1. https://twitter.com/Conaw/status/1240104928661475328
2. https://github.com/athensresearch/athens/discussions/829

**Comment by juniusfree** [comment]
Some ideas:
1. `[new content](((block-id)))`
2. `(block-id|alias)` https://discord.com/channels/708122962422792194/708375112537342025/785287221506736149
3. whatever the syntax, add to `/` command like block-embed

**Reply by yifanyin** [reply]
Upvote for 2nd idea. Can either of these be applied to the page alias, too?

**Reply by jefftangx** [reply]
1. Is more of an extension of markdown: []() brackets, parens
2. Is more typical for wiki-style syntaxes: | pipe

I'm pretty sure | works well for blocks, because blocks already have a concept of block ids, a random string. I'm less sure for page links because in general, we can have ANY character inside [[]] and it is treated as a page. Exception might be another pair of [[]] for nested links, and / for namespaced pages (which we haven't invested in deeply yet)

Should do some research on how other apps are doing it to understand the trade offs more deeply. This actually extends into a deeper question of aliases.

**Reply by shanberg** [reply]
Two thoughts...

1. The more metadata/relationships/extras we attach to blocks/pages, the more sense it makes to provide a separate UI to define those things, like writing on the backside of an index card instead of trying to fit all the semantics into the main block content.
2. Is it possible to start trending away from lots and lots of nested parens and brackets? The syntax just feels really heavy and hard to read.

**Reply by jefftangx** [reply]
> The more metadata/relationships/extras we attach to blocks/pages, the more sense it makes to provide a separate UI to define those things, like writing on the backside of an index card instead of trying to fit all the semantics into the main block content.

Wait I love the index card analogy. I think [Mem](https://get.mem.ai/) and RemNotes leans harder into this. Roam does the `::` syntax for properties, e.g. 

* block
  * prop:: value

but I don't think this scales well. Though it doesn't enable their attribute table feature, a very interesting but still prototype-y feature.

> Is it possible to start trending away from lots and lots of nested parens and brackets? The syntax just feels really heavy and hard to read.

Yeah I don't like the complexity of nested parens/brackets. I feel that no one actually uses `[[nested [[links]]` are really not usable after one level of nesting. Also why I wanna be thoughtful with new syntax. Lots of overhead to learn, then very difficult to maintain compatability.

**Reply by neotyk** [reply]
Meta overlay for blocks?
We could show what we've recognized in syntax: block ref, embeds, page links.
Though would it really provide any value, all above are clearly expressed in rendering.

With new parser implementing `[alias](((block-id)))` should be trivial,  
also `((block-id|alias))` would cause much problem.
Indeed amount of parens in 1st is a bit much.
How about shorter version for *Athens Flavoured Markdown*: `[alias]((block-id))` ?

**Reply by neotyk** [reply]
Do we want to get rid of nested page links?
👍

**Reply by jefftangx** [reply]
I don't think we need to get rid of them. Not harmful to have it for now.

**Comment by jefftangx** [comment]
We implemented aliases this way: https://github.com/athensresearch/athens/discussions/1326#discussioncomment-923738
